lib/ex_unit/lib/ex_unit/capture_io.ex

```diff
@@ -141,6 +154,29 @@ defmodule ExUnit.CaptureIO do
     end
   end
 
+  defp do_capture_io(pid, :standard_io, options, fun) do
+    prompt_config = Keyword.get(options, :capture_prompt, true)
+    input = Keyword.get(options, :input, "")
+
+    original_gl = Process.group_leader()
+    {:ok, capture_gl} = StringIO.open(input, capture_prompt: prompt_config)
+    case ExUnit.CaptureServer.process_capture_on(pid) do
+      {:ok, ref} ->
+        try do
+          Process.group_leader(pid, capture_gl)
+          do_capture_io(capture_gl, fun)
+        after
+          if Process.alive?(pid) do
+            Process.group_leader(pid, original_gl)
```

Comment:
Seems we're setting incorrect group leader here, leader of the called but not of the `pid`. 🤔